### PR TITLE
fix: set correct schema on config import

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

When we import examples the schema is set to `null`, even in databases that support schemas and have a default schema. This is problematic, because users might create duplicate physical datasets, which are usually not allowed, resulting in, eg:

- `examples.NULL.birth_names` (actually in the default "public" schema, for Postgres)
- `examples.public.birth_names`

These duplicate datasets have caused problems in the past, because they break assumptions about physical datasets.

This PR changes the `load-examples` command and the import mechanism to explicitly set the schema to the default one when one is not set.

There's a tricky case that this PR needs to handle: when a user creates a dataset duplicating one of the new [examples defined in YAML files](https://github.com/apache/superset/tree/master/superset/examples/configs/datasets/examples). These datasets have fixed UUIDs, and the following happens:

1. Admin runs `superset load-examples`.
2. Dataset `users` is created with `table_name=users`, `schema=null` and `uuid=7195db6b-2d17-7619-b7c7-26b15378df8c`.
3. Later a user creates a physical dataset pointing to `users`. When adding a dataset we force the user to select a schema, so this dataset has `table_name=users`, `schema=public` (in Postgres) and a random UUID. Normally you can't create duplicate physical datasets, but because the schema is technically different this works.
4. Admin runs `superset load-examples` again, after merging this PR.
5. The importer sees that the `users` dataset matches the one with NULL schema, based on the UUID. it tries to update that dataset, setting its schema to "public".
6. This fails because there's already a dataset with the same `database_id`, `table_name` and `schema` (the one created by the user).

This is a rare case, but it can happen. I added a workaround so that when this happens the duplicate datasets are not updated. This could be potentially problematic in the future, if we add more columns to a dataset and have charts use those columns; then the charts would fail to render because the dataset wouldn't be updated.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

1. Run `superset load-examples` on `master`.
2. Datasets have no schema.
3. Run `superset load-examples` on this branch.
4. Datasets have a schema if appropriate (for Postgresl, "public").

Same for clean installations. I tested with Postgresql, MySQL, and SQLite.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/16051
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/commands/importers/v1/examples.py [hunk 1]
```diff
@@ -103,6 +104,14 @@ def _import(
                 else:
                     config["database_id"] = database_ids[config["database_uuid"]]
 
+                # set schema
+                if config["schema"] is None:
+                    url = make_url(
+                        conf.get("SQLALCHEMY_EXAMPLES_URI")
+                        or conf.get("SQLALCHEMY_DATABASE_URI")
```
Comment (john-bodley) [inline superset/commands/importers/v1/examples.py:111]:
```suggestion
                        or conf["SQLALCHEMY_DATABASE_URI"]
```

By construction this is guaranteed to exist.

Also @betodealmeida per [here](https://github.com/apache/superset/blob/master/superset/__init__.py#L36-L39) the `conf` field is deemed to be legacy, i.e., we likely should be using  `current_app.config` instead.

Comment (betodealmeida) [reply superset/commands/importers/v1/examples.py]: Thanks, @john-bodley! I'll update to use `current_app.config` instead.

**Review by mistercrunch — APPROVED** [review]

**Review by eschutho — APPROVED** [review]
any way in the future to dry up some of the duplication?

**Comment by junlincc** [comment]
@betodealmeida Hi Beto! 
do we have plan to merge this PR?

**Pull request merged**